### PR TITLE
Small improvements batch

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/scan/ProjectLock.java [hunk 1]
```diff
@@ -55,13 +56,25 @@ public void tryLock() {
...
   }
-  
+
+  private static void hideLockFileWindows(Path p) {
+    try {
+      DosFileAttributeView fileAttrView = Files.getFileAttributeView(p, DosFileAttributeView.class);
+      if (fileAttrView != null) {
+        fileAttrView.setHidden(true);
+      }
+    } catch (IOException e) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/scan/ProjectLock.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java [hunk 2]
```diff
@@ -138,14 +150,25 @@ protected void handleIgnoreEoL(char c) {
...
+      }
+    }
+
+    private void processBuffer() {
+      try {
+        if (sb.length() > 0) {
+          ByteBuffer encoded = encoder.encode(CharBuffer.wrap(sb));
+          globalMd5Digest.update(encoded.array(), 0, encoded.limit());
+        }
+      } catch (CharacterCodingException e) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java [hunk 3]
```diff
@@ -182,10 +209,21 @@ protected void newLine() {
...
     }
 
+    private void processBuffer() {
+      try {
+        if (sb.length() > 0) {
+          ByteBuffer encoded = encoder.encode(CharBuffer.wrap(sb));
+          lineMd5Digest.update(encoded.array(), 0, encoded.limit());
+          consumer.consume(line, lineMd5Digest.digest());
+        }
+      } catch (CharacterCodingException e) {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 3 critical

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM appart comment about duplicate module name log.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 2 critical

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 2 critical

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 2 critical

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Protected method 'toProperties(...)' is never used.](https://github.com/SonarSource/sonarqube/blob/6ef6d49565762721184d75ccc22f2244149d658b/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectReactorBuilder.java#L266) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUnusedProtectedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'java.io.FileInputStream'.](https://github.com/SonarSource/sonarqube/blob/d753685361d795beba1e32dda369989e410c5fa5/sonar-batch/src/main/java/org/sonar/batch/scan/ProjectReactorBuilder.java#L30) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Pull request closed without merging**